src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Ball {
+    private String color;
+    private int number;
+
+    public Ball() {
+        this.color = new ColorSupplier().getRandomColor();
+        this.number = new Random().nextInt(100);
+    }

```

Comment:
separation of concerns design principle - ball model itself shouldn't be bothered with logic of generating random properties, it should be independent. This logic should be in lottery class instead. Ball should allow its users to create a ball given some values. Why limit yourself to the only possible usage scenario? Create a constructor which takes two arguments

read where `this` is needed - https://stackoverflow.com/a/2411283